### ISSUE TITLE
Enhancement: Allow fetching data sources by name and id

### ISSUE THREAD

**Issue by smithjw** [deploymenttheory/terraform-provider-jamfpro#484]
At the current time, if I want to create data-sources for multiple resources of a given type, I need to create a single resource per item, specified by id. In the case of multiple environments, this can be tedious as the id will likely be different between environments. If I was instead able to find the resource by name, this would be much easier as the name is identical across environments. 

A real-world example of this is that I'm soon going to be managing the same Jamf tenant (containing the environments dev, staging, & prod) within multiple repositories. In the primary repo, I'll be managing shared resources like Categories and Extension Attributes, but in a separate repo, I'll have the terraform code to create Policies/Profiles/Groups for a particularly department that they can manage independently.

The Jamf Pro API allows us to find Extension Attributes in three ways, id, name, or all EAs. Is it possible to update the `jamfpro_computer_extension_attribute` & `jamfpro_category` data sources to allow finding by name? Alternatively, could a new data source (eg `jamfpro_computer_extension_attributes`), fetch all EAs in a given instance and list them out in a map?

**Comment by ShocOne** [comment]
Hi there,

I believe your approach would be prone to stating issues if the name was to ever change. so im reluctant to implement something like that.

in terms of adding a list to return all id's and names of a given resource type, that's something that could be considered. i've poc'd something for webhooks

where

```
data "jamfpro_webhook_list" "example" {}

output "webhook_ids" {
  value = data.jamfpro_webhook_list.example.ids
}

output "webhooks" {
  value = data.jamfpro_webhook_list.example.webhooks
}
```

would give the following

![image](https://github.com/user-attachments/assets/26cda438-e003-4cbd-a109-81c55009d0ec)

is this what you had in mind ?

thanks

**Comment by smithjw** [comment]
Yeah, I think your approach of just outputting a list with ID/Name, is the best path forward here. In terms of using the API endpoints that can output all resources of a given type, the main ones I'd use initially are:
- Webhooks
- Categories
- Extension Attributes
- Scripts
- Packages

**Comment by smithjw** [comment]
Fetching by name is working really well in my sub repos. I'm currently utilising it for the following and can confirm they work as expected:

- Categories
- Extension Attributes
- Smart Computer Groups
- Packages
- Scripts

One thing I'm curious when it comes to fetching packages in the data source via name, is it possible to match on glob patterns, eg returning an array of items that match?

**Comment by ShocOne** [comment]
Awesome. Thats good to hear.  We are going to extend this pattern to all of the tf resource types over time. Could you describe in a little more detail what problem you're trying to solve by returning an array of matches based upon a string ?

**Comment by ShocOne** [comment]
Hi @smithjw , could you provide an update else I'll consider this ticket closed. Thanks

**Comment by smithjw** [comment]
Actually I think I'm less wanting to return everything within an array/list, but more being able to get a package by name that includes a version number that may increment over time.

**Comment by ShocOne** [comment]
Version is not a field that exists within the packages resource.  So you will need to come up with another way of storing the version value and handling it, perhaps in the notes field? 

Closing the ticket.
